src/solvers/adaptive_time_solver.C

```diff
@@ -64,6 +65,16 @@ void AdaptiveTimeSolver::init()
   // needs to handle new vectors, diff_solver->init(), etc
   core_time_solver->init();
 
+  // Set the core_time_solver's solution history object to be the same one as
+  // that for the outer adaptive time solver
+  core_time_solver->set_solution_history(*(this->get_solution_history()));
+
+  // Now that we have set the SolutionHistory object for the coretimesolver,
+  // we set the SolutionHistory type for the timesolver to be NoSolutionHistory
+  // All storage and retrieval will be handled by the coretimesolver directly.
+  NoSolutionHistory outersolver_solution_history;
+  this->set_solution_history(outersolver_solution_history);
```

Comment:
Since we're doing this reset anyway, plus retaining the use of SolutionHistory::clone(), do we still need to change solution_history to be a shared_ptr?  We didn't end up actually sharing it.